prow/pod-utils/gcs/upload.go

```diff
@@ -149,48 +151,70 @@ func FileUploadWithOptions(file string, opts pkgio.WriterOptions) UploadFunc {
 				*opts.BufferSize = 25 * 1024 * 1024
 			}
 		}
+		if err := reader.Close(); err != nil {
+			return err
+		}
 
-		uploadErr := DataUploadWithOptions(reader, opts)(writer)
+		newReader := func() (io.ReadCloser, error) {
+			reader, err := os.Open(file)
```

Comment:
missed this in the initial review. os.Open is invoked twice in FileUploadWithOptions, and the first call at line 142 is just for Stat(), which makes sense with the old logic as the reader was returns for reading later. I think we can replace line 142 with os.Info, and since there is no reader you won't need to Close() either.

Reply:
That's correct, thanks!